src/oph/heratepalvelu/amis/AMISherateHandler.clj

```diff
@@ -30,7 +30,8 @@
                        (check-organisaatio-whitelist?
                          koulutustoimija
                          (date-string-to-timestamp (:alkupvm herate)))
-                       (check-sisaltyy-opiskeluoikeuteen? opiskeluoikeus))
+                       (check-sisaltyy-opiskeluoikeuteen? opiskeluoikeus)
+                       (check-valid-herate-date (:alkupvm herate)))
```

Comment:
Olisiko mahdollista sijoittaa tämä ennen kaikkia (tai ennen mahdollisimman monia) checkejä, joissa tehdään kyselyjä muihin palveluihin tai tietokantaan? Jos ei haluta muodostaa linkkiä koska päivämäärä on väärä, kannattaa näin päättää ennen kuin haetaan opiskeluoikeus.